src/main/java/com/lightstep/tracer/shared/TextMapPropagator.java

```diff
@@ -16,53 +18,40 @@
     static final String FIELD_NAME_SAMPLED = PREFIX_TRACER_STATE + "sampled";
 
     public void inject(SpanContext spanContext, final TextMap carrier) {
-        carrier.put(FIELD_NAME_TRACE_ID, Long.toHexString(spanContext.getTraceId()));
-        carrier.put(FIELD_NAME_SPAN_ID, Long.toHexString(spanContext.getSpanId()));
+        carrier.put(FIELD_NAME_TRACE_ID, Util.toHexString(spanContext.getTraceId()));
+        carrier.put(FIELD_NAME_SPAN_ID, Util.toHexString(spanContext.getSpanId()));
         carrier.put(FIELD_NAME_SAMPLED, "true");
         for (Map.Entry<String, String> e : spanContext.baggageItems()) {
             carrier.put(PREFIX_BAGGAGE + e.getKey(), e.getValue());
         }
     }
 
     public SpanContext extract(TextMap carrier) {
-        int requiredFieldCount = 0;
-
-        // traceId and spanId initial values are guaranteed to be overwritten because of the
-        // requiredFieldCount check
-        long traceId = 0;
-        long spanId = 0;
-        Map<String, String> decodedBaggage = null;
+        Long traceId = null;
+        Long spanId = null;
+        Map<String, String> baggage = new HashMap<>();
```

Comment:
Not a real comment, just want to note that reading this refactor makes me laugh, code was pretty convoluted for no reason, 👍  to improvements!

Reply:
😄